### PR TITLE
use BytesIO on py2 for speeeed

### PR DESCRIPTION
**Pull request by blampe**
`six.BytesIO` is `StringIO` on python 2.

@abhinav @breerly @junchaowu

### CODE REVIEW

**Comment by junchaowu** [comment]
image

**Comment by junchaowu** [comment]
Do we have concern about the Unicode part?

**Comment by blampe** [comment]
@junchaowu I think at this layer everything should be utf8-encoded bytes but @abhinav can confirm.

File: CHANGELOG.rst [hunk 1]
```diff
@@ -4,7 +4,7 @@ Releases
 0.4.1 (unreleased)
 ------------------
 
-- Nothing changed yet.
+- Now uses ``io.BytesIO`` for speed improvements on Python 2.
```
Comment (abhinav) [inline CHANGELOG.rst]: nit: in Python 2

**Comment by abhinav** [comment]
:+1:

**Pull request merged**